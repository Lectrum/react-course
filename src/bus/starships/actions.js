// Types
import { types } from './types';

import { api } from '../../api';

export const starshipsActions = Object.freeze({
    // Sync
    fill: (payload) => {
        return {
            type: types.STARSHIPS_FILL,
            payload
        }
    },
    // Async
    fetchAsync: () => async (dispatch) => {
        dispatch({
            type: types.STARSHIPS_FETCH_ASYNC
        });
        
        const response = await api.starships.fetch();
        const { results } = await response.json();
        
        dispatch(starshipsActions.fill(results));
    }
});