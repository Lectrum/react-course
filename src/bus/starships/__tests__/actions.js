import { starshipsActions } from "../actions";
import expectExport from "expect";

describe("starships actions:", () => {
  test("startFetching", () => {
    expectExport(starshipsActions.startFetching()).toMatchSnapshot();
  });
  test("stopFetching", () => {
    expectExport(starshipsActions.stopFetching()).toMatchInlineSnapshot(`
      Object {
        "type": "STARSHIPS_STOP_FETCHING",
      }
    `);
  });
  test("fill", () => {
    expectExport(starshipsActions.fill()).toMatchSnapshot();
  });
  test("fetchAsync", () => {
    expectExport(starshipsActions.fetchAsync()).toMatchSnapshot();
  });
});
