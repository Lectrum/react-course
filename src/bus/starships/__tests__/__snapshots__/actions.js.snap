// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`starships actions: fetchAsync 1`] = `[Function]`;

exports[`starships actions: fill 1`] = `
Object {
  "payload": undefined,
  "type": "STARSHIPS_FILL",
}
`;

exports[`starships actions: startFetching 1`] = `
Object {
  "type": "STARSHIPS_START_FETCHING",
}
`;
